src/Symfony/Component/Form/ChoiceList/Factory/CachingFactoryDecorator.php

```diff
@@ -101,9 +114,31 @@ public function createListFromChoices(iterable $choices, $value = null)
     /**
      * {@inheritdoc}
      */
-    public function createListFromLoader(ChoiceLoaderInterface $loader, $value = null)
+    public function createListFromLoader($loader, $value = null)
```

Comment:
BC break here isn't it?

Reply:
I guess you're right... I need the handle the case where this decorator is extended 🤔,  what about annotating factories as final? They should be decorated instead.

Reply:
Fixed.